src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/Microsoft.Diagnostics.Monitoring.UnitTests.csproj

```diff
@@ -10,7 +10,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <Compile Include="..\..\Microsoft.Diagnostics.Monitoring.RestServer\CorsConfiguration.cs" Link="Options\CorsConfiguration.cs" />
```

Comment:
This (the CorsConfiguration) is linked in from the rest assembly.

Reply:
Are you asking why not use the type from the REST assembly? The unit test project compiles for both ASP.NET Core 3.1 and 5.0; I don't think I can reference the REST assembly on the 5.0 compilation due to FrameworkReference differences.

Reply:
That makes sense, but makes me wonder about the dotnet-monitor reference? Is that one purely for build ordering?

Reply:
Thinking about it some more, do we want to make this test 3.1 only? I realize that the test could really be anything, but if we want to reuse the types in restserver/dotnet-monitor, making the test 3.1 would make it easier.

Reply:
If I changed it to be only 3.1, I'd have to figure out a way to run each test for dotnet-monitor running as 3.1 and 5.0. I want the tests to flex the runtime roll-forward capabilities so we aren't blindsided with an incompatibility. Also, the test app that I will want to write should run as both 3.1 and 5.0 so we can test different polyfill capabilities (cookie generation, process information gathering, `/logs` behavior, etc). As it is right now, the tests build for both 3.1 and 5.0 and they run dotnet-monitor (and eventually the test app) with the same runtime version. Then it'll get even more complicated when I go to add 6.0 previews.

So I could change it to just be 3.1, but I think I'd have to convert the Fact tests to be Theory tests that pull in the runtime versions so that each test is run for each runtime version.

Reply:
Alternative: put all of the shared types in a new netstandard2.0 library (one for REST assembly, one for dotnet-monitor)

Alternative: put all of the shared types into shared projects (does not produce assembly, but effectively links in types into each dependent assembly).

Reply:
I'm going to keep the changes as-is for now and then revisit this after thinking about it and discussing it more.